internal/controllers/core/kubernetesapply/reconciler.go

```diff
@@ -142,7 +143,8 @@ func (r *Reconciler) shouldDeployOnReconcile(
 	nn types.NamespacedName,
 	ka *v1alpha1.KubernetesApply,
 	imageMaps map[types.NamespacedName]*v1alpha1.ImageMap) bool {
-	if ka.ObjectMeta.Labels[v1alpha1.LabelOwnerKind] == v1alpha1.LabelOwnerKindTiltfile {
+	owner := metav1.GetControllerOf(ka)
+	if owner != nil && owner.Kind == v1alpha1.OwnerKindTiltfile {
```

Comment:
I feel like this is a pretty cool milestone!

Reply:
YAS